pkg/types/builder.go

```diff
@@ -447,8 +447,11 @@ func (r *resource) addReferenceFields(g *Builder, paramName *types.TypeName, fie
 // generateTypeName generates a unique name for the type if its original name
 // is used by another one. It adds the former field names recursively until it
 // finds a unique name.
-func generateTypeName(suffix string, pkg *types.Package, names ...string) (string, error) {
+func generateTypeName(suffix string, pkg *types.Package, overrideFieldNames map[string]string, names ...string) (string, error) {
 	n := names[len(names)-1] + suffix
+	if v, ok := overrideFieldNames[n]; ok {
+		return v, nil
+	}
```

Comment:
nit: So, my understanding is that we would like to override any auto-generated name that we calculate here with the manually configured (overridden) name. If this is the case, then shouldn't we override what's calculated at the end of this function because there are some further modifications down the path in this function. The idea is:
- We bump an API version
- Run `make generate` and observe a parameter name conflict
- At this point, it would be easier for us to use the exact conflicting name as the key.
 
If we do the look up early here, then the override key may not match the actual conflicting name. What do you think?

Reply:
Makes sense. This implementation resolves the problems until we observed. However, your suggestion is more powerful and generic. Thanks @ulucinar!

Reply:
Thank you @sergenyalcin.